src/MOI/MOI_wrapper.jl

```diff
@@ -1682,6 +1682,14 @@ function MOI.add_constraint(
         end
     end
     Xpress.chgcoltype(model.inner, [info.column], Cchar['B'])
+    if info.type == CONTINUOUS
```

Comment:
Might be worth adding a comment above this, along the lines of `chgcoltype` resets variable bounds to `[0, 1]`, so we need to re-set them if they have been changed.